tests/unit/credentials/CredentialsManager.js

```diff
@@ -0,0 +1,191 @@
+const async = require('async');
+const assert = require('assert');
+const sinon = require('sinon');
+const { Logger } = require('werelogs');
+
+const { VaultServiceCredentials } =
+    require('../../../lib/credentials/VaultServiceCredentials');
+const AccountCredentials = require('../../../lib/credentials/AccountCredentials');
+const CredentialsManager = require('../../../lib/credentials/CredentialsManager');
+const { authTypeAccount, authTypeVault } = require('../../../lib/constants');
+
+const extension = 'test-extension';
+const id = '5a4847c9d83443885282cf38a09fa1a6193de9ed970168376a7b10c438683da0';
+const accountId = '000000000000';
```

Comment:
Having a second account id would be a good idea to ensure that the code always separates creds by account ids.